// react-native-increment-decrement-button/dist/commonjs/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactNative = require("react-native");

let {
  height
} = _reactNative.Dimensions.get('window');

let {
  width
} = _reactNative.Dimensions.get('window');

const styles = _reactNative.StyleSheet.create({
  viewOuter: {
    flexDirection: 'row'
  },
  viewBtnLeft: {
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: 18,
    height: _reactNative.PixelRatio.roundToNearestPixel(height * 5 / 100),
    width: _reactNative.PixelRatio.roundToNearestPixel(width * 8 / 100),
    borderWidth: 1,
    borderColor: '#eeeeee',
    borderTopLeftRadius: _reactNative.PixelRatio.roundToNearestPixel(height * 1 / 100),
    borderBottomLeftRadius: _reactNative.PixelRatio.roundToNearestPixel(height * 1 / 100)
  },
  viewBtnRight: {
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: '#eeeeee',
    height: _reactNative.PixelRatio.roundToNearestPixel(height * 5 / 100),
    width: _reactNative.PixelRatio.roundToNearestPixel(width * 8 / 100),
    borderTopRightRadius: _reactNative.PixelRatio.roundToNearestPixel(height * 1 / 100),
    borderBottomRightRadius: _reactNative.PixelRatio.roundToNearestPixel(height * 1 / 100)
  },
  viewTextInput: {
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: 18,
    borderWidth: 1,
    borderColor: '#eeeeee',
    height: _reactNative.PixelRatio.roundToNearestPixel(height * 5 / 100),
    width: _reactNative.PixelRatio.roundToNearestPixel(width * 8 / 100)
  },
  labelStyle: {
    fontSize: 10
  }
});

var _default = styles;
exports.default = _default;
